compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1575,69 +1574,66 @@ void OMR::Power::MemoryReference::accessStaticItem(TR::Node *node, TR::SymbolRef
       {
       TR::Node *topNode = cg->getCurrentEvaluationTreeTop()->getNode();
       TR::UnresolvedDataSnippet *snippet = NULL;
-      int32_t  tocIndex = symbol->getStaticSymbol()->getTOCIndex();
-
-      if (cg->comp()->compileRelocatableCode())
+      int32_t  tocIndex = cg->comp()->isOutOfProcessCompilation() ? PTOC_FULL_INDEX : symbol->getStaticSymbol()->getTOCIndex();
```

Comment:
Perhaps we should consider making `TR_PPCTableOfConstants` an extensible class and changing its behaviour in OpenJ9? That way you wouldn't need checks like this or as many changes to the OMR TOC class, it could mostly be self-contained downstream? Either that or introduce a more elegant way for downstream to request that the TOC be disabled throughout the compiler. Having lots of little changes dealing with returning `PTOC_FULL_INDEX` sprinkled everywhere looks distasteful.

Reply:
This is actually the only place where I've had to do something like this (and I added it as a precautionary measure). I don't think this is needed either, actually. 

PTOC_FULL_INDEX is defined as 0 in the code base:
```
#define PTOC_FULL_INDEX       0
```
The static symbol constructors initialize the TOC index to `0` as well:
```
   StaticSymbol(TR::DataType d) :
      TR::Symbol(d),
      _staticAddress(0),
      _assignedTOCIndex(0)
      {
      _flags.setValue(KindMask, IsStatic);
      }

   StaticSymbol(TR::DataType d, void * address) :
      TR::Symbol(d),
      _staticAddress(address),
      _assignedTOCIndex(0)
      {
      _flags.setValue(KindMask, IsStatic);
      }

   StaticSymbol(TR::DataType d, uint32_t s) :
      TR::Symbol(d, s),
      _staticAddress(0),
      _assignedTOCIndex(0)
      {
      _flags.setValue(KindMask, IsStatic);
      }
```

The `_assignedTOCIndex` field above is `private` and there is only one use of the `setTOCIndex` routine that writes to it:
```
      if (tocIndex == 0 && !symbol->isStartPC())
         {
         tocIndex = TR_PPCTableOfConstants::lookUp(ref, cg);
         symbol->getStaticSymbol()->setTOCIndex(tocIndex);
         }
```
After https://github.com/eclipse/omr/pull/4443/ is merged, the `TR_PPCTableOfConstants::lookUp(ref, cg)` routine used above will return `PTOC_FULL_INDEX`. So the `symbol->getStaticSymbol()->getTOCIndex()` should never return anything other than `PTOC_FULL_INDEX` anyway.

I'm going to remove this. After this, the only place where we do these kinds of checks and return `PTOC_FULL_INDEX` will be inside the `TR_PPCTableOfConstants` class.